job_executor/config/log.py

```diff
@@ -1,114 +1,105 @@
+import sys
 import json
 import logging
 import logging.handlers
-import sys
-import traceback
-from datetime import datetime, UTC
-from multiprocessing import Queue
-
-from json_logging import util
+import datetime
 import tomlkit
+import threading
 
+from typing import Tuple
+from multiprocessing import Queue
 from job_executor.config import environment
 
 
-class ContextFilter(logging.Filter):
-    """
-    This is a filter which injects contextual information into the log.
-    """
+def _get_project_meta():
+    with open("pyproject.toml", encoding="utf-8") as pyproject:
+        file_contents = pyproject.read()
+    return tomlkit.parse(file_contents)["tool"]["poetry"]
+
+
+class MicrodataJSONFormatter(logging.Formatter):
+    def __init__(self):
+        self.pkg_meta = _get_project_meta()
+        self.host = environment.get("DOCKER_HOST_NAME")
+        self.command = json.dumps(sys.argv)
+
+    def format(self, record: logging.LogRecord) -> str:
+        # When receiving logs from the queue and passing them to
+        # the log handler the record is wrapped in another record
+        # and placed in the msg field.
+        # When this is the case we log the original record
+        if isinstance(record.msg, logging.LogRecord):
+            record = record.msg
+        return json.dumps(
+            {
+                "@timestamp": datetime.datetime.fromtimestamp(
+                    record.created,
+                    tz=datetime.timezone.utc,
+                ).strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3]
+                + "Z",
+                "command": self.command,
+                "error.stack": record.__dict__.get("exc_info"),
+                "host": self.host,
+                "message": record.getMessage(),
+                "level": record.levelno,
+                "levelName": record.levelname,
+                "loggerName": record.name,
+                "schemaVersion": "v3",
+                "serviceName": "job-executor",
+                "serviceVersion": str(self.pkg_meta["version"]),
+                "thread": record.threadName,
+            }
+        )
+
 
+class WorkerFormatter(logging.Formatter):
     job_id = ""
 
     def __init__(self, job_id: str):
-        super().__init__()
         self.job_id = job_id
 
-    def filter(self, record):
-        # prefix record.msg instead of adding a new field
-        # to be compliant with Kibana
-        record.msg = self.job_id + ": " + str(record.msg)
-        return True
-
-
-def configure_worker_logger(queue: Queue, job_id: str):
-    queue_handler = logging.handlers.QueueHandler(queue)
-    queue_handler.setLevel(logging.INFO)
-    # use default formatter here so that the JSON formatter will be
-    # applied when reading from queue
-    queue_handler.formatter = logging.Formatter()
+    def format(self, record: logging.LogRecord) -> str:
+        record.msg = f"{self.job_id}: {record.msg}"
+        return record
```

Comment:
The `format` method should return a `str` ([ref](https://docs.python.org/3/library/logging.html#logging.Formatter.format)). Maybe use a filter like the previous version of the code?